tests/test_eman_source.py

```diff
@@ -0,0 +1,524 @@
+            (691, 3101),
+            (2387, 3125),
+            (1219, 3149),
+            (2977, 3151),
+            (2299, 3167),
+            (1737, 3193),
+            (2037, 3283),
+            (715, 3201),
+            (993, 3197),
+            (3901, 3227),
+            (2163, 3247),
+            (861, 3253),
+            (499, 3301),
+            (1345, 3311),
+            (225, 3245),
+            (1941, 3299),
+            (747, 3299),
+            (3221, 3295),
+            (1495, 3309),
+            (1235, 3297),
+            (3335, 3315),
+            (2727, 3333),
+            (1077, 3315),
+            (2287, 3335),
+            (3107, 3341),
+            (1191, 3375),
+            (1705, 3403),
+            (343, 3379),
+            (3351, 3393),
+            (1869, 3385),
+            (3763, 3403),
+            (827, 3401),
+            (2769, 3405),
+            (1805, 3423),
+            (3079, 3441),
+            (925, 3509),
+            (1187, 3457),
+            (3367, 3463),
+            (379, 3471),
+            (3007, 3499),
+            (1917, 3529),
+            (2501, 3491),
+            (725, 3491),
+            (2327, 3477),
+            (3331, 3485),
+            (2391, 3509),
+            (1087, 3501),
+            (2701, 3553),
+            (3593, 3533),
+            (2011, 3561),
+            (583, 3559),
+            (2581, 3575),
+            (2131, 3587),
+            (1565, 3571),
+            (3435, 3577),
+            (1579, 3577),
+            (3763, 3583),
+            (3943, 3579),
+            (2403, 3619),
+            (505, 3611),
+            (697, 3609),
+            (1023, 3621),
+            (355, 3623),
+            (3593, 3653),
+            (327, 3673),
+            (3273, 3685),
+            (2797, 3701),
+            (1127, 3697),
+            (3819, 3711),
+            (2657, 3719),
+            (2829, 3753),
+            (3059, 3739),
+            (3159, 3763),
+            (525, 3769),
+            (2993, 3771),
+            (1169, 3799),
+            (1507, 3785),
+            (1733, 3781),
+            (2655, 3847),
+            (2053, 3861),
+            (2855, 3883),
+        }
+        self.tmpdir = tempfile.mkdtemp()
+        # get path to test .mrc file
+        with importlib_resources.path(tests.saved_test_data, "sample.mrc") as test_path:
+            self.mrc_path = str(test_path)
+        # create a coord file (will only contain centers) and a box file (will contain lower left corner as well as X and Y dimensions of particle). These are two ways the coordinate information is provided, so we need to test both. also several cases of bad files.
+        self.coord_fp = os.path.join(self.tmpdir, "sample.coord")
+        self.box_fp = os.path.join(self.tmpdir, "sample.box")
+        self.box_fp_nonsquare = os.path.join(self.tmpdir, "sample_nonsquare.box")
+        # populate box and coord files
+        with open(self.coord_fp, "w") as coord, open(self.box_fp, "w") as box, open(
```

Comment:
is this confusing context line just to avoid looping over centers for each case? Why not just three clearer loops?

Reply:
yes. will replace by three loops